### PR TITLE
Add ability to use a variable as filter.

### LINKED ISSUE #133: Using a variable as filter does not work
The following test is failing

GraphQL-Query:

```graphql
query filterQuery($filter: _PersonFilter) { person(filter: $filter) { name }}
```

Query variables

```json
{
  "filter": {
    "name": "Jane",
    "company": {
      "name_ends_with": "ME"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by Andy2003**
resolves #133

### CODE REVIEW

**Comment by Andy2003** [comment]
@jexp we need to merge `1.0`-branch to `master` after this PR is merged

File: src/main/kotlin/org/neo4j/graphql/handler/QueryHandler.kt [hunk 1]
```diff
@@ -97,7 +97,7 @@ class QueryHandler private constructor(
         } else {
             "($variable:${label()})"
         }
-        val where = where(variable, fieldDefinition, type, propertyArguments(field), field)
+        val where = where(variable, fieldDefinition, type, propertyArguments(field), field, env)
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/handler/QueryHandler.kt:100]: I don't think we should pass `env` through to an inner method but just the parameters/variables

File: src/main/kotlin/org/neo4j/graphql/handler/projection/ProjectionBase.kt [hunk 2]
```diff
@@ -99,11 +105,16 @@ open class ProjectionBase {
             .toMap()
     }
 
-    private fun filterExpressions(value: Any?, type: GraphQLFieldsContainer): List<Predicate> {
-        // todo variable/parameter
-        return if (value is Map<*, *>) {
-            CompoundPredicate(value.map { (k, v) -> toExpression(k.toString(), v, type) }, "AND").parts
-        } else emptyList()
+    private fun filterExpressions(value: Any?, type: GraphQLFieldsContainer, variables: MutableMap<String, Any>): List<Predicate> {
```
Comment (jexp) [inline src/main/kotlin/org/neo4j/graphql/handler/projection/ProjectionBase.kt:108]: can we use Map<> here not MutableMap

File: src/test/resources/filter-tests.adoc [hunk 3]
```diff
@@ -1837,37 +1837,39 @@ RETURN graphql.labels(company) AS _labels,
 
 === Filter as variable
 
-CAUTION: Is not working
-
 .GraphQL-Query
 [source,graphql]
```
Comment (jexp) [inline src/test/resources/filter-tests.adoc:1841]: do we also have a test where inside of a filter a variable/parameter is used?

**Review by jexp — CHANGES_REQUESTED** [review]
see my comments

File: src/test/resources/filter-tests.adoc (hunk 3 above)
Comment (Andy2003) [reply src/test/resources/filter-tests.adoc:1841]: Do you mean the `Variable pass through`-Test?

**Review by jexp — APPROVED** [review]

**Pull request merged**